src/workers/notifications/commentDeleted.ts

```diff
@@ -0,0 +1,45 @@
+import { messageToJson, Worker } from '../worker';
+import { Notification, Comment } from '../../entity';
+import { ChangeObject } from '../../types';
+
+interface Data {
+  comment: ChangeObject<Comment>;
+}
+
+const worker: Worker = {
+  subscription: 'api.comment-deleted-notification-cleanup',
+  handler: async (message, con, logger) => {
+    const data: Data = messageToJson(message);
+    const { comment } = data;
+
+    try {
+      await con
+        .getRepository(Notification)
+        .createQueryBuilder()
+        .delete()
+        .where({
+          referenceType: 'comment',
+          referenceId: comment?.id,
```

Comment:
This is risky. I suggest validating that id exists before, otherwise we may risk deleting all comment notifications